src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -233,8 +248,88 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
                                   path,
                                   nvc: context!.ParsedQueryString,
                                   after);
-            rootEnumerated = rootEnumerated.Append(nextLink);
-            return OkResponse(JsonSerializer.SerializeToElement(rootEnumerated));
+
+            rootEnumerated = RemoveExtraFieldsInResponseWithMultipleItems(rootEnumerated.ToList(), extraFieldsInResponse);
+            return OkResponse(JsonSerializer.SerializeToElement(rootEnumerated.Append(nextLink)));
+        }
+
+        /// <summary>
+        /// To support pagination and $first clause with Find requests, it is necessary to provide the nextLink
+        /// field in the response. For the calculation of nextLink, the fields such as primary keys, fields in $orderby clause
+        /// are retrieved from the database in addition to the fields requested in the $select clause.
+        /// However, these fields are not required in the response.
+        /// This function helps to determine those additional fields that are present in the response.
+        /// </summary>
+        /// <param name="response">Response json retrieved from the database</param>
+        /// <param name="context">FindRequestContext for the GET request.</param>
+        /// <returns>Additional fields that are present in the response</returns>
+        private static IEnumerable<string> DetermineExtraFieldsInResponse(JsonElement response, FindRequestContext context)
```

Comment:
will there always be extra fields in a response when the count of returned items is >= 1? Is there no way to short circuit some of these loops if no $first, $after, $select query string params exist?